File: 2022--蛇优化(Snake Optimizer, SO)/MATLAB/main.m
%___________________________________________________________________%
%  Snake Optimizer (SO) source codes version 1.0                    %
%                                                                   %
%  Developed in MATLAB R2021b                                       %
%                                                                   %
%  Author and programmer:  Fatma Hashim & Abdelazim G. Hussien      %
%                                                                   %
%         e-Mail: fatma_hashim@h-eng.helwan.edu.eg                  %
%                 abdelazim.hussien@liu.se                          %
%                 aga08@fayoum.edu.eg                               %
%                                                                   %
%                                                                   %
%   Main paper: Fatma Hashim & Abdelazim G. Hussien                 %
%               Knowledge-based Systems                             %
%               in press,                                           %
%               DOI: 10.1016/j.knosys.2022.108320                   %
%                                                                   %
%___________________________________________________________________%
close all
clear all
clc
fitfun = @Chung_Reynolds;
dim=30;
Max_iteration=1000;
SearchAgents_no=30;
lb=-100;
ub=100;
tlt='Chung Reynolds';
i=1;
[Xfood, Xvalue,CNVG] = SO(SearchAgents_no,Max_iteration,fitfun, dim,lb,ub)
figure,
plot(CNVG,'Color', 'r')
xlim([1 1000]);

File: 2022--蛇优化(Snake Optimizer, SO)/PYTHON/main.py
from utils import  SO
from matplotlib import  pyplot as plt

fitfun = 'Chung_Reynolds'
dim=30
Max_iteration=1000
SearchAgents_no=30
lb=-100
ub=100
tlt='Chung Reynolds'
i=1
Xfood, Xvalue,CNVG = SO(SearchAgents_no,Max_iteration,fitfun, dim,lb,ub)

plt.figure(figsize=(10,10))
plt.plot(range(Max_iteration),CNVG)
plt.xlim([1,1000])
plt.show()